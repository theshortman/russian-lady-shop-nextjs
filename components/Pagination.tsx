import { FC } from "react";
import Link from "next/link";

interface PaginationProps {
  prevPageLink: string | null;
  nextPageLink: string | null;
}

const Pagination: FC<PaginationProps> = ({ prevPageLink, nextPageLink }) => {
  return (
    <nav aria-label="Page navigation">
      <ul className="pagination justify-content-center">
        {prevPageLink && (
          <li className="page-item">
            <Link href={prevPageLink}>
              <a className="page-link" aria-label="Previous">
                Пред.
              </a>
            </Link>
          </li>
        )}

        {nextPageLink && (
          <li className="page-item">
            <Link href={nextPageLink}>
              <a className="page-link" aria-label="Next">
                След.
              </a>
            </Link>
          </li>
        )}
      </ul>
    </nav>
  );
};

export default Pagination;
