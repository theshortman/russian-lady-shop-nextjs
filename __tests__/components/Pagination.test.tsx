import Pagination from "@/components/Pagination";
import { render, screen } from "@testing-library/react";

describe("<Pagination />", () => {
  test("renders correctly if links are not null", () => {
    const { container } = render(
      <Pagination
        prevPageLink="/test-prev-link"
        nextPageLink="/test-next-link"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("prev link should not be in the document if it is null", () => {
    render(<Pagination prevPageLink={null} nextPageLink={null} />);
    expect(screen.queryByText("Пред.")).not.toBeInTheDocument();
  });

  test("next link should not be in the document if it is null", () => {
    render(<Pagination prevPageLink={null} nextPageLink={null} />);
    expect(screen.queryByText("След.")).not.toBeInTheDocument();
  });
});
