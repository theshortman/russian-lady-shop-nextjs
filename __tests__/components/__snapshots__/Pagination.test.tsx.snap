// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders correctly if links are not null 1`] = `
<div>
  <nav
    aria-label="Page navigation"
  >
    <ul
      class="pagination justify-content-center"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="Previous"
          class="page-link"
          href="/test-prev-link"
        >
          Пред.
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next"
          class="page-link"
          href="/test-next-link"
        >
          След.
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
